js_modules/dagit/packages/core/src/asset-graph/useLiveDataForAssetKeys.tsx

```diff
@@ -27,13 +34,126 @@ export function useLiveDataForAssetKeys(assetKeys: AssetKeyInput[]) {
     return liveResult.data ? buildLiveData(liveResult.data) : {};
   }, [liveResult.data]);
 
+  // Track whether the data is being refetched so incoming asset events don't trigger
+  // duplicate requests for live data.
+  const fetching = React.useRef(false);
+  fetching.current = [NetworkStatus.refetch, NetworkStatus.loading].includes(
+    liveResult.networkStatus,
+  );
+
+  const timerRef = React.useRef<NodeJS.Timeout | null>(null);
+
+  const onRefreshDebounced = React.useCallback(() => {
+    // This is a basic `throttle`, except that if it fires and finds the live query
+    // is already refreshing it debounces again.
+    const refetch = liveResult.refetch;
+    const fire = () => {
+      if (fetching.current) {
+        timerRef.current = setTimeout(fire, SUBSCRIPTION_MAX_POLL_RATE);
+      } else {
+        timerRef.current = null;
+        refetch();
+      }
+    };
+    if (!timerRef.current) {
+      timerRef.current = setTimeout(fire, SUBSCRIPTION_MAX_POLL_RATE);
+    }
+    return () => {
+      if (timerRef.current) {
+        clearTimeout(timerRef.current);
+      }
+    };
+  }, [timerRef, liveResult.refetch]);
```

Comment:
Slightly confused by this -- the `callback` prop appears to be `() => void`, but this `useCallback` returns a function.

Reply:
Oh good catch, it looks like this is a mashup of a useEffect and a useCallback. Will fix!